crates/fuel-core/src/database.rs

```diff
@@ -325,35 +325,23 @@ where
     }
 }
 
-impl Modifiable for Database<OnChain> {
-    fn commit_changes(&mut self, changes: Changes) -> StorageResult<()> {
-        commit_changes_with_height_update(self, changes, |iter| {
-            iter.iter_all::<FuelBlocks>(Some(IterDirection::Reverse))
-                .map(|result| result.map(|(height, _)| height))
-                .try_collect()
-        })
-    }
-}
-
-impl Modifiable for Database<OffChain> {
-    fn commit_changes(&mut self, changes: Changes) -> StorageResult<()> {
-        commit_changes_with_height_update(self, changes, |iter| {
-            iter.iter_all::<FuelBlockIdsToHeights>(Some(IterDirection::Reverse))
-                .map(|result| result.map(|(_, height)| height))
-                .try_collect()
-        })
-    }
+macro_rules! impl_modifiable_for_database {
```

Comment:
I like the clear name, but it's kinda misleading since it doesn't explain why we only use this macro for some DBs, and not others e.g. `GenesisDatabase<OnChain>`.

The design of the macro is driven by DRY rather than maintainability/readability. In other words, the reader of the code needs to read other code to understand why this macro is being used. Whereas the less-DRY code is uncoupled in that sense.

Reply:
My bad, I'll close it.